shared-bindings/adcbuffer/__init__.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * SPDX-FileCopyrightText: Copyright (c) 2022 Lee Atkinson, MeanStride Technology, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <stdint.h>
+#include "py/obj.h"
+#include "py/runtime.h"
+#include "shared-bindings/microcontroller/Pin.h"
+#include "shared-bindings/adcbuffer/__init__.h"
+#include "shared-bindings/adcbuffer/BufferedInput.h"
+
+//| """Analog buffered hardware support
+//|
+//| The `adcbuffer` module contains classes to provide access to analog-to-digital
+//| conversion and digital-to-analog (DAC) for multiple value transfer.
```

Comment:
You mention DAC, but it's not using a DAC. Did you meant that? If that's a future possibility (`BufferedOutput`), then let's call the whole module `analogbufferedio` or something like that.

Reply:
So, I agree that this should be more than ADC and also include DAC for other platforms. You also had suggested bufferedanalogio. I took a poll and it seems it reads better and is preferred over analogbufferedio. So I will be renaming the module to bufferedanalogio with BufferedIn with read() and BufferedOut with write(). Hope that agrees with most.

Reply:
How about `analogbufio` or (more awkward sounding) `bufanalogio`? I like the first because it starts with `analog`, and `buf` is good enough (`buffered` is pretty long).

Reply:
While I like bufferedanalogio better than analogbufferedio (like broken red car over red broken car) I also agree its quite long. I like starting it with `analog` too and `bufio` has a kind of mening to most of us anyway. So, anaolgbufio it is .... going once, going twice, .....